snapr/src/lib.rs

```diff
@@ -73,12 +76,56 @@ pub type TileFetcher<'a> =
 /// }
 /// ```
 #[cfg(not(feature = "rayon"))]
-pub type TileFetcher<'a> = &'a dyn Fn(i32, i32, u8) -> Result<image::DynamicImage, Error>;
+pub type IndividualTileFetcher<'a> = &'a dyn Fn(i32, i32, u8) -> Result<image::DynamicImage, Error>;
+
+/// Function that takes in a slice of coordinates and a zoom level as arguments and returns the [`Images`](image::DynamicImage) of the map tiles at the given positions.
+///
+/// ## Example
+///
+/// ```rust
+/// use image::DynamicImage;
+///
+/// fn tile_fetcher(matrix: &[(i32, i32)], zoom: u8) -> Result<Vec<(i32, i32, DynamicImage)>, snapr::Error> {
+///     todo!()
+/// }
+/// ```
+// FIXME: `BatchTileFetcher` does not truly require `Send` or `Sync` when the `rayon` feature flag is enabled.
+// It's only here currently to get the `Snapr::overlay_backing_tiles::x_y_to_tile` closure to compile correctly.
```

Comment:
I have other priorities at the moment or I wouldn't leave this here. It is what is for the time being.